examples/custom_provider/file_provider.c

```diff
@@ -0,0 +1,354 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+#define _GNU_SOURCE 1
+
+#include "umf/base.h"
+#include "umf/pools/pool_scalable.h"
+#include "umf/providers/provider_os_memory.h"
+
+#include <fcntl.h>
+#include <linux/falloc.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <unistd.h>
+
+// Define the size for address reservation
+#define ADDRESS_RESERVATION ((size_t)16 * 1024 * 1024 * 1024)
+
+// Macro to align a value up to the nearest multiple of align
+#define ALIGN_UP(value, align) (((value) + (align)-1) & ~((align)-1))
+
+// Struct to represent the file provider
+typedef struct file_provider_t {
+    void *ptr;        // Pointer to the reserved memory
+    size_t offset;    // Offset for the next allocation
```

Comment:
nit: poffset? "Offset in memory for ..."

Reply:
Done